spec/features/movies/show_spec.rb

```diff
@@ -0,0 +1,91 @@
+    @movie_2 = Movie.create!(title: "Tangled", creation_year: "2000", genre: "animation", studio: @studio_1)
+    @movie_3 = Movie.create!(title: "Babe", creation_year: "1997", genre: "farm", studio: @studio_2)
+    @movie_4 = Movie.create!(title: "IDK", creation_year: "1975", genre: "mystery", studio: @studio_2)
+
+    @actor_1 = Actor.create!(name: "Sam", age: 32)
+    @actor_2 = Actor.create!(name: "Max", age: 33)
+    @actor_3 = Actor.create!(name: "Ben", age: 30)
+  end
+
+  describe 'When a user visits a movies show page, there is information' do
+    it 'They see the movies title, creation year and genre' do
+      visit "movies/#{@movie_1.id}"
+
+      expect(page).to have_content(@movie_1.title)
+      expect(page).to have_content(@movie_1.creation_year)
+      expect(page).to have_content(@movie_1.genre)
+    end
+
+    it 'They see actors in order from youngest to oldest, and average age' do
+      @movie_1.actors << @actor_1
+      @movie_1.actors << @actor_2
+      @movie_1.actors << @actor_3
+
+      visit "movies/#{@movie_1.id}"
+      expect(page).to have_content(@movie_1.actor_age_avg)
+
+      oldest = find("#actor-#{@actor_2.id}")
+      mid = find("#actor-#{@actor_1.id}")
+      youngest = find("#actor-#{@actor_3.id}")
+
+      expect(youngest).to appear_before(mid)
+      expect(mid).to appear_before(oldest)
+
+      within "#actor-#{@actor_1.id}" do
+        expect(page).to have_content(@actor_1.name)
+        expect(page).to have_content(@actor_1.age)
+      end
+
+      within "#actor-#{@actor_2.id}" do
+        expect(page).to have_content(@actor_2.name)
+        expect(page).to have_content(@actor_2.age)
+      end
+
+      within "#actor-#{@actor_3.id}" do
+        expect(page).to have_content(@actor_3.name)
+        expect(page).to have_content(@actor_3.age)
+      end
+    end
+  end
+
+  describe 'When a user visits movie show page, they can add an actor' do
+    it 'They do not see any actors not part of this movie' do
+      @movie_1.actors << @actor_1
+      visit "/movies/#{@movie_1.id}"
+
+      expect(page).to have_content(@actor_1.name)
+      expect(page).to_not have_content(@actor_2.name)
+      expect(page).to_not have_content(@actor_3.name)
+    end
+
+    it 'They see a form to add an actor to this movie (by id)' do
+      visit "/movies/#{@movie_1.id}"
+
+      expect(page).to have_selector(:link_or_button, "Add Actor")
+    end
+
+    it "They fill in the form with a actor id, clicks button, redirects back to show page, and see actor name" do
+      visit "/movies/#{@movie_1.id}"
+      expect(page).to_not have_content(@actor_1.name)
+      
+      visit "/movies/#{@movie_1.id}"
+      fill_in("Actor", with: @actor_1.id)
+      click_button("Add Actor")
+
+      expect(current_path).to eq("/movies/#{@movie_1.id}")
+      
+      within "#actor-#{@actor_1.id}" do
+        expect(page).to have_content(@actor_1.name)
+      end
+    end
+  end
+  
+end
```

Comment:
These are great tests and I love your use of `within` blocks.

I only have one suggestion for the future to further reduce the ambiguity by using more specific assertions. Here's what I mean by that. This assertion, `expect(page).to have_content(@movie_1.creation_year)` just says that `1995` just needs to show up on the page somewhere, it could be part of a date, a phone number, etc. It doesn't specify that it's the year of the movie at all.

This could be resolved by placing this inside a `within` block or specifying a more specific assertion like: `expect(page).to have_content("Year: #{@movie_1.creation_year}")`. While this may seem like a small thing, it helps reduce ambiguity and helps avoid false positives by testing exactly what and where you intend to test.